src/ray/util/internal/stream_redirection_utils.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright 2025 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/util/internal/stream_redirection_utils.h"
+
+#include <memory>
+#include <utility>
+
+#include "ray/util/compat.h"
+
+namespace ray::internal {
+
+RedirectionHandleWrapper::RedirectionHandleWrapper(MEMFD_TYPE_NON_UNIQUE stream_fd,
```

Comment:
```suggestion
StreamRedirectionHandle::StreamRedirectionHandle(MEMFD_TYPE_NON_UNIQUE stream_fd,
```
"Wrapper" does not really convey any meaning. The suggested naming is very obvious to me (good) and it matches `StreamRedirectionOption` directly.

Reply:
totally makes sense, updated!